iceoryx_examples/README.md

```diff
@@ -22,7 +22,8 @@ cd someExample
 
 | Example                                                | Description |Level |
 |:-------------------------------------------------------|:------------|:----:|
-|[callbacks](./callbacks/)                           | Implement callbacks which are triggered by events. | Intermediate |
+|[callbacks](./callbacks/)                               | Implement callbacks which are triggered by events. | Intermediate |
+|[callbacks_in_c](./callbacks_in_c/)                     | The C version of the callbacks example. | Intermediate |
```

Comment:
Please include the Readme of the new example in `doc/website/getting-started/examples/intermediate-examples.md`

Reply:
Please include `callbacks_in_c` in this list too:
https://github.com/eclipse-iceoryx/iceoryx/blob/8dd43130e7abf99fd30044b3763595338743ac9a/tools/iceoryx_build_test.sh#L286

This is for the out-of-tree build in the Github CI

Reply:
I think I will forget this on a constant basis. I will also add a hint to the general example readme on what to do if a new example is added.
@mossmaurice @dkroenke is there anything else to do?

Reply:
@elfenpiff Great list! I think once #540 is merged another item would be "Write a integration test for your example". Then it'll be run every time the CI runs